types/figlet/index.d.ts

```diff
@@ -0,0 +1,423 @@
+        | 'Star Strips'
+        | 'Star Wars'
+        | 'Stellar'
+        | 'Stforek'
+        | 'Stick Letters'
+        | 'Stop'
+        | 'Straight'
+        | 'Stronger Than All'
+        | 'Sub-Zero'
+        | 'Swamp Land'
+        | 'Swan'
+        | 'Sweet'
+        | 'THIS'
+        | 'Tanja'
+        | 'Tengwar'
+        | 'Term'
+        | 'Test1'
+        | 'The Edge'
+        | 'Thick'
+        | 'Thin'
+        | 'Thorned'
+        | 'Three Point'
+        | 'Ticks Slant'
+        | 'Ticks'
+        | 'Tiles'
+        | 'Tinker-Toy'
+        | 'Tombstone'
+        | 'Train'
+        | 'Trek'
+        | 'Tsalagi'
+        | 'Tubular'
+        | 'Twisted'
+        | 'Two Point'
+        | 'USA Flag'
+        | 'Univers'
+        | 'Varsity'
+        | 'Wavy'
+        | 'Weird'
+        | 'Wet Letter'
+        | 'Whimsy'
+        | 'Wow'
+        ;
+
+    type KerningMethods =
+        | 'default'
+        | 'full'
+        | 'fitted'
+        | 'controlled smushing'
+        | 'universal smushing'
+        ;
+    type PrintDirection = number;
+
+    interface FittingRules {
+        vLayout: number;
+        vRule5: boolean;
+        vRule4: boolean;
+        vRule3: boolean;
+        vRule2: boolean;
+        vRule1: boolean;
+        hLayout: number;
+        hRule6: boolean;
+        hRule5: boolean;
+        hRule4: boolean;
+        hRule3: boolean;
+        hRule2: boolean;
+        hRule1: boolean;
+    }
+
+    interface Options {
+        font?: Fonts;
+        horizontalLayout?: KerningMethods;
+        verticalLayout?: KerningMethods;
+        printDirection?: PrintDirection;
+        showHardBlanks?: boolean;
+    }
+    interface FontOptions {
+        hardBlank: string;
+        height: number;
+        baseline: number;
+        maxLength: number;
+        oldLayout: number;
+        numCommentLines: number;
+        printDirection: PrintDirection;
+        fullLayout: number | null;
+        codeTagCount: number | null;
+        fittingRules: FittingRules;
+    }
+    interface Defaults {
+        font: Fonts;
+        fontPath: string;
+    }
+
+    interface TextMethod {
+        (
+            txt: string,
+            cb: (error: Error | null, result?: string) => void,
+        ): void;
+        (
+            txt: string,
+            font: Fonts,
+            cb: (error: Error | null, result?: string) => void,
+        ): void;
+        (
+            txt: string,
+            // tslint:disable-next-line: unified-signatures
+            options: Options | undefined,
+            cb: (error: Error | null, result?: string) => void,
+        ): void;
+    }
+
+    interface TextSyncMethod {
+        (
+            txt: string,
+            font?: Fonts,
+        ): string;
+        (
+            txt: string,
+            // tslint:disable-next-line: unified-signatures
+            options: Options,
+        ): string;
+    }
+
+    interface Figlet extends TextMethod {
+        text: TextMethod;
```

Comment:
Instead of making this as interface here write the function declaration at the location where you have const figlet: figlet.FigLet

And all the methods in the interface are exported function declarations in the namespace

Reply:
@sheetalkamat Why is that preferred? Could you let me know the reason?

Reply:
@sheetalkamat I originally do this to reuse same type to define `text` and root function.

Reply:
It is to make it more readable and follow the template

Reply:
@sheetalkamat Is there a way to reuse the type of root function? Do I have to write the same type repeatedly?